src/cachemap.h

```diff
@@ -105,8 +105,7 @@ class CacheMap
 
     bool Insert(const K& key, const V& value)
     {
-        map_it it = mapIndex.find(key);
-        if(it != mapIndex.end()) {
+        if(mapIndex.find(key) != mapIndex.end()) {
```

Comment:
This .find() call can be eliminated completely if you rely on the return value of the .insert() later as it will inform you if insertion failed due to an already existing item. That'd require to swap the calls to push_front() and insert().

Reply:
Not sure if this can be done here - `listItems` shouldn't be updated is there is a duplicate but we need actual `listItems.begin()` to insert into `mapIndex` which we can't get without pushing an item. Can be mitigated by erasing an item from the list if `mapIndex.insert()` fails but this doesn't sound right.

Reply:
Ahh true, overlooked this problem.